### ISSUE TITLE
offline sync of cloud archives

### ISSUE THREAD

**Issue by CircleCode** [buttercup/buttercup-desktop#307]
It would be interesting to allow certain archives to be synchronized for offline usage.
For example, my archive is on a webdav server.
Sometimes, i do not have any web connection. But I'd like to use my passwords.
On another side, i need these passwords across several devices, thus they need to shared online.
What i'd love is to mark this webdav archive as available offline: each time I access it with web connection, it synchronizes, and is used offline hen network is not available.
If I understood it correctly, it already has merging features, so this wouldn't be an issue.

**Comment by perry-mitchell** [comment]
That's a good point.. but I wonder: at this point, wouldn't it be more likely that a user is using a sync tool? If you're using Dropbox/ownCloud/Nextcloud/Seafile/Yandex etc., there's likely a sync client which you might've installed.

If we're to implement such a feature in light of the other option we'd need to handle all the syncing ourselves.. I personally think that there should be some community interest in such a feature before implementing it. This is a good idea, but having it as a feature would mean extra maintenance and testing, so I'm hesitant..

**Comment by CircleCode** [comment]
I understand the points you highlighted, and I agree with you : this would increase code complexity, decrease maintainability, and can probably be solved by using cloud systems already doing synchronisation.

But in this case, there is no more merge feature.

The fact is that I would like to use buttercup on my computer and my phone. So I need a shared archive. what happens if my computer is offline, I update a password, and while my computer is still offline, I update a password on my phone?
In fact, the merge feature already implemented in buttercup is only about race conditions, right?

**Comment by perry-mitchell** [comment]
> what happens if my computer is offline, I update a password, and while my computer is still offline, I update a password on my phone?

Yeah this is a good point. Ultimately offline usage is important, and is something we need to consider sooner rather than later. I think we should consider putting this on the roadmap, but it'd need core support. @sallar - opinions mate?

**Comment by Aktariel** [comment]
Speaking as someone who uses their computer and phone in areas with little or no service on a regular enough basis that this would be critical, I agree. Nextcloud/Owncloud/Dropbox/etc, all have local storage when offline, but the scenario as described would require some sort of merge/conflict management.

**Comment by Luraktinus** [comment]
Any news on this?
